api/serializers.py

```diff
@@ -1,43 +1,61 @@
-import base64
-
-from django.contrib.auth.models import User
-from django.core.files.base import ContentFile
-from django.utils.timezone import now
+from django.contrib.auth import authenticate
+from django.contrib.auth.models import User, update_last_login
 from rest_framework import serializers
 from rest_framework_jwt.settings import api_settings
 
-from .models import Video
+from .models import UserProfile, Video
+
+JWT_PAYLOAD_HANDLER = api_settings.JWT_PAYLOAD_HANDLER
+JWT_ENCODE_HANDLER = api_settings.JWT_ENCODE_HANDLER
 
 
 class UserSerializer(serializers.ModelSerializer):
     class Meta:
-        model = User
-        fields = ("username",)
+        model = UserProfile
+        fields = ("first_name", "last_name", "phone_number", "age", "gender")

```

Comment:
This is a breaking change and causes the tests to fail. The matching view is expecting this to serialize a `User`, not a `UserProfile`. Either change the matching view to use `UserProfile` and fix the test or leave the `UserSerializer` as-is and make a `UserProfileSerializer`. I was planning on making the switch to `UserProfile`s once they were added since they have the actual data I need to construct the matches on the frontend.